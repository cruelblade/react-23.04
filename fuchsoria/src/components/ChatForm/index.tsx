import React, { useState, useRef } from 'react';
import { Button, Input } from 'antd';
import { IChatFormProps } from '../../interfaces';
import styles from './styles.module.scss';

export default function ChatForm({ handleSubmit }: IChatFormProps) {
  const [message, setMessage] = useState('');
  const messageRef = useRef<Input | null>(null);

  const sendMessage = (e: React.SyntheticEvent) => {
    e.preventDefault();
    messageRef.current?.focus();

    if (message) {
      handleSubmit(message);
      setMessage('');
    }
  };

  return (
    <form onSubmit={sendMessage} className={styles.chatForm}>
      <div className={styles.chatFormGroup}>
        <Input
          placeholder="Your Message"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
          ref={messageRef}
        />
        <Button type="primary" htmlType="submit">
          Send Message
        </Button>
      </div>
    </form>
  );
}
