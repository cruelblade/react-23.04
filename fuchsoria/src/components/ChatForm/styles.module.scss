.chat-form {
  padding: 10px;
  background-color: #d6d6d6;

  &__group {
    display: flex;
    align-items: center;
  }
}
