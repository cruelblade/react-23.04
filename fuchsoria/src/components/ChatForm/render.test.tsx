import React from 'react';
import { render } from '@testing-library/react';
import ChatForm from '../ChatForm';

describe('<ChatForm />', () => {
  it('Component should render', () => {
    const { container } = render(<ChatForm handleSubmit={jest.fn} />);

    expect(container).toMatchSnapshot();
  });
});
