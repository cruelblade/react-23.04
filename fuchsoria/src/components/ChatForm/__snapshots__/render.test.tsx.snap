// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatForm /> Component should render 1`] = `
<div>
  <form
    class="chatForm"
  >
    <div
      class="chatFormGroup"
    >
      <input
        class="ant-input"
        placeholder="Your Message"
        type="text"
        value=""
      />
      <button
        class="ant-btn ant-btn-primary"
        type="submit"
      >
        <span>
          Send Message
        </span>
      </button>
    </div>
  </form>
</div>
`;
